libs/components/src/lib/popup/popup.spec.ts

```diff
@@ -228,20 +228,44 @@ describe('vwc-popup', () => {
 
 	describe('handle keydown', () => {
 		it('should hide on escape key', async () => {
-			const anchor = await setAnchor();
-			element.anchor = 'anchor';
+			const anchor = await setupPopupToOpenWithAnchor();
+			expect(element.open)
+				.toEqual(true);
+
+			await elementUpdated(element);
+			anchor.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }));
 			await elementUpdated(element);
 
-			element.open = true;
+			expect(element.open)
+				.toEqual(false);
+		});
+
+		it('should remove keydown listener after disconnection', async function() {
+			const anchor = await setupPopupToOpenWithAnchor();
 			expect(element.open)
 				.toEqual(true);
 
+			await elementUpdated(element);
+			element.disconnectedCallback();
 			await elementUpdated(element);
 			anchor.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }));
+
+			expect(element.open)
+				.toEqual(true);
+		});
```

Comment:
```suggestion
		it('should remove keydown listener after disconnection', async function() {
			const anchor = await setupPopupToOpenWithAnchor();
			const openStateBeforeEsc = element.open;

			await elementUpdated(element);
			element.disconnectedCallback();
			await elementUpdated(element);
			anchor.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }));

      expect(openStateBeforeEsc)
				.toEqual(true);
			expect(element.open)
				.toEqual(true);
		});
```

Convention is to set the expectations at the end of the test. Make sure it is the same for all other tests plz.